pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -128,16 +129,41 @@ func ConvertPolicyToJwtConfig(policy *authn.Policy) *jwtfilter.JwtAuthentication
 		AllowMissingOrFailed: true,
 	}
 	for _, policyJwt := range policyJwts {
-		hostname, port, _, err := model.ParseJwksURI(policyJwt.JwksUri)
-		if err != nil {
-			log.Errorf("Cannot parse jwks_uri %q: %v", policyJwt.JwksUri, err)
-			continue
+		jwt := &jwtfilter.JwtRule{
+			Issuer:               policyJwt.Issuer,
+			Audiences:            policyJwt.Audiences,
+			ForwardPayloadHeader: OutputLocationForJwtIssuer(policyJwt.Issuer),
+			Forward:              true,
 		}
 
-		jwt := &jwtfilter.JwtRule{
-			Issuer:    policyJwt.Issuer,
-			Audiences: policyJwt.Audiences,
-			JwksSourceSpecifier: &jwtfilter.JwtRule_RemoteJwks{
+		for _, location := range policyJwt.JwtHeaders {
+			jwt.FromHeaders = append(jwt.FromHeaders, &jwtfilter.JwtHeader{
+				Name: location,
+			})
+		}
+		jwt.FromParams = policyJwt.JwtParams
+
+		if useInlinePublicKey {
+			jwtPubKey, err := model.JwtKeyResolver.GetPublicKey(policyJwt.JwksUri)
+			if err != nil {
+				log.Warnf("Failed to fetch jwt public key from %q", policyJwt.JwksUri)
+			}
+
+			// Put empty string in config even if above ResolveJwtPubKey fails.
+			jwt.JwksSourceSpecifier = &jwtfilter.JwtRule_LocalJwks{
+				LocalJwks: &core.DataSource{
+					Specifier: &core.DataSource_InlineString{
+						InlineString: base64.StdEncoding.EncodeToString([]byte(jwtPubKey)),
```

Comment:
https://github.com/istio/proxy/pull/1482 removed decoding, so you can remove encoding here too.

Reply:
done